### ISSUE TITLE
Image in product gallery doesn't have a focus outline indicator

### ISSUE THREAD

**Issue by Manussakis** [woocommerce/storefront#2187]
## Describe the bug
There is no focus outline indicator when the link that wraps the image in the product gallery is focused on.
This issue fails on the [Focus Visible (Level AA) WCAG](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html) criterion.

Isolating the problem (mark completed items with an [x]):
- [x] I have deactivated other plugins and themes and confirmed this bug occurs when only WooCommerce + Storefront theme are active.
- [x] I can reproduce this bug consistently using the steps below.

## To Reproduce
1. Go to a product page.
2. Tab navigate through the page until the focus stops on the link that wraps the image in the gallery.
3. No focus outline indicator is visible.

## Screenshots
No outline focus indicator is visible.
![Image](https://github.com/user-attachments/assets/5b4e7b4c-9e04-4d42-ba9d-73776102d7aa)

## Expected behavior
The link should get an outline focus when focused:
![Image](https://github.com/user-attachments/assets/e90e4e60-9105-46cc-a6df-30ca69c6fb14)

### Browser Environment
Please provide as much detail as possible about your testing environment.

- Platform: macOS
- Browser(s): Chrome, Firefox, and Safari

### WordPress Environment

<details>
```
WordPress address (URL): http://woocommerce.test
Site address (URL): http://woocommerce.test
WC Version: 9.9.0-dev
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.9.2
Log Directory Writable: ✔
WP Version: 6.8.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –
```
</details>
